catalog_tools/analysis/estimate_beta.py

```diff
@@ -48,12 +48,46 @@ def estimate_beta_utsu(magnitudes: np.ndarray, mc: float, delta_m: float = 0
     Returns:
         beta:       maximum likelihood beta (b_value = beta * log10(e))
     """
-    beta = 1 / (np.mean(magnitudes) - mc + delta_m / 2)
+    beta = 1 / np.mean(magnitudes - mc + delta_m / 2)
 
     return beta
 
 
-def estimate_beta_elst(magnitudes: np.ndarray) -> float:
+def differences(magnitudes: np.ndarray) -> np.ndarray:
+    """returns all the differences between the magnitudes.
+
+    Args:
+        magnitudes: vector of magnitudes differences, sorted in time (first
+                    entry is the earliest earthquake)
+
+    Returns: array of all differences of the elements of the input
+    """
+    mag_diffs = np.array([])
+    for ii, mag in enumerate(magnitudes):
+        loop_mag = np.delete(magnitudes, [ii], axis=0)
+        mag_diffs = np.append(mag_diffs, loop_mag - mag)
+    return mag_diffs
+
+
+def differences_successive(magnitudes: np.ndarray) -> np.ndarray:
```

Comment:
not sure I mentioned this already, but this whole function is already implemented in numpy.diff(), so you don't need to define it again.